### PR TITLE
[DOP-3638]: Transpile API handlers from TS to JS

### PR DESCRIPTION
**Pull request by branberry**
## Ticket
[DOP-3638](https://jira.mongodb.org/secure/RapidBoard.jspa?rapidView=953&projectKey=DOP&view=detail&selectedIssue=DOP-3638)

## Notes

For the AWK CDK project, we now need to manually bundle the TypeScript files that exist within the `api` directory to JavaScript. I have taken the additional step to zip these files, so that we reduce the bundle size when we deploy the lambda functions. 

Given that the API files have TypeScript errors, I decided to resolve these in this PR. This is because we will run into build failures in our CI/CD pipeline when we perform the build command.

## Dependency changes
I added a few additional dependencies for typing purposes. Here is a list of the new dependencies:

```json
    "aws-lambda": "^1.0.7",
    "@octokit/webhooks-types": "^6.11.0",
    "@types/aws-lambda": "^8.10.114"
```

### CODE REVIEW

File: api/controllers/v1/jobs.ts
Comment (branberry) [inline api/controllers/v1/jobs.ts:184]: removed this return value as it is never used anywhere

File: api/controllers/v1/jobs.ts
Comment (branberry) [inline api/controllers/v1/jobs.ts:287]: removed this return value as it is never used anywhere

File: src/entities/job.ts
Comment (branberry) [inline src/entities/job.ts:58]: I made this null because in the `prepGithubPushPayload` function, we assign null to this value. This then gets upserted into the collection, so it appears that we have been providing a null value for this one previously.

File: api/controllers/v1/jobs.ts
Comment (branberry) [inline api/controllers/v1/jobs.ts:161]: This isn't used, but I am not sure if there is value in keeping it around still. Any thoughts on keeping this function around?

File: api/controllers/v1/github.ts
Comment (branberry) [inline api/controllers/v1/github.ts:43]: The `JobStatus.inQueue` enum has `'inQueue'` as the string value.

File: api/controllers/v1/jobs.ts
Comment (schmalliso) [reply api/controllers/v1/jobs.ts:143]: I think there is little value in keeping it - we used to try all failed builds three times (I don't recall why) but it was very annoying and provided no value. I'm pro removing unused code.

File: src/entities/job.ts
Comment (schmalliso) [inline src/entities/job.ts:28]: It's probably out of scope for this, but we should really remove this at some point since we don't have different sizes of builders anymore and *everything* is Xlarge.

**Review by schmalliso — APPROVED** [review]
This looks great and also I want another set of eyes that aren't mine before you merge 😁

File: src/entities/job.ts
Comment (branberry) [reply src/entities/job.ts:28]: I'll remove, it's actually a small change. I did a quick search, and we only ever set this value to true when we assign it a value at all.

File: api/controllers/v1/jobs.ts
Comment (mmeigs) [inline api/controllers/v1/jobs.ts:17]: Is this connection ever used? (I know this doesn't have to do with your PR, but great time to clean this if so)

**Review by mmeigs — APPROVED** [review]
All this typing work is just so appreciated. LGTM. One possible additional deletion below, but all is good.

File: api/controllers/v1/jobs.ts
Comment (branberry) [reply api/controllers/v1/jobs.ts]: Good catch! It is not used. I will remove.

**Pull request merged**
